t/ui/16-tests_job_next_previous.t

```diff
@@ -276,5 +276,38 @@ subtest 'bug reference shown' => sub {
     );
 };
 
+subtest 'server-side limit has precedence over user-specified limit' => sub {
+    my $limits = OpenQA::App->singleton->config->{misc_limits};
+    $limits->{next_ajax_max_limit} = 5;    # set maximum limit for next
+    $limits->{previous_ajax_max_limit} = 5;    # set maximum limit for previous
+    $limits->{next_ajax_default_limit} = 2;    # set default limit for next
+    $limits->{previous_ajax_default_limit} = 2;    # set default limit for previous
+
+    $t->get_ok('/tests/99910/ajax?previous_limit=0&next_limit=6', 'query with exceeding user-specified limit for next')
+      ->status_is(200);
+    my $jobs = $t->tx->res->json->{data};
+    is ref $jobs, 'ARRAY', 'data returned (1)' and is scalar @$jobs, 7, 'maximum limit for next is effective';
```

Comment:
Why are 7 jobs expected here? The limit should max it out at 5 "next" jobs. Then there's supposedly still the current job returned. So 6 jobs would make sense. Is the 7th job a previous job one always gets (despite the previous limit being 0)? It would likely be good to have an explaining comment. I think the same counts for the other checks.

Reply:
A good question that leads to this [messy query](https://github.com/os-autoinst/openQA/blob/ec2d97adf17caf5a4aabb4f51a995abca885a86b/lib/OpenQA/Schema/Result/JobNextPrevious.pm#L20), which i've never seen before. Wonder how that's not a method yet.

Reply:
Yes, by default the latest and the current job are always returned, plus the 5 (if the limit is effective) equals 7

Reply:
Ah, the latest. That makes sense as well. It would still be good to put that information in a comment.

Reply:
i've added `# expected amount is the defined limit plus current and latest job`